zebra-rpc/src/methods.rs

```diff
@@ -611,19 +640,62 @@ where
                         message: error.to_string(),
                         data: None,
                     })?;
-
-                match response {
+                let tx = match response {
                     zebra_state::ReadResponse::TransactionIdsForBlock(Some(tx_ids)) => {
-                        let tx_ids = tx_ids.iter().map(|tx_id| tx_id.encode_hex()).collect();
-                        Ok(GetBlock::Object { tx: tx_ids })
+                        tx_ids.iter().map(|tx_id| tx_id.encode_hex()).collect()
                     }
                     zebra_state::ReadResponse::TransactionIdsForBlock(None) => Err(Error {
                         code: MISSING_BLOCK_ERROR_CODE,
                         message: "Block not found".to_string(),
                         data: None,
-                    }),
+                    })?,
                     _ => unreachable!("unmatched response to a transaction_ids_for_block request"),
-                }
+                };
+
+                // Get block confirmations from the block height index
+                //
+                // # Concurrency
+                //
+                // All possible responses are valid, even if a block is added to the chain, or
+                // the best chain changes. Clients must be able to handle chain forks, including
+                // different confirmation values before or after added blocks, and switching
+                // between -1 and multiple different confirmation values.
+
+                // From <https://zcash.github.io/rpc/getblock.html>
+                const NOT_IN_BEST_CHAIN_CONFIRMATIONS: i64 = -1;
+
+                let confirmations = if let Some(hash) = hash_or_height.hash() {
+                    let request = zebra_state::ReadRequest::Depth(hash);
+                    let response = state
+                        .ready()
+                        .and_then(|service| service.call(request))
+                        .await
+                        .map_err(|error| Error {
+                            code: ErrorCode::ServerError(0),
+                            message: error.to_string(),
+                            data: None,
+                        })?;
+
+                    match response {
+                        // Confirmations are one more than the depth.
+                        // Depth is limited by height, so it will never overflow an i64.
+                        zebra_state::ReadResponse::Depth(Some(depth)) => Some(i64::from(depth) + 1),
+                        zebra_state::ReadResponse::Depth(None) => {
+                            Some(NOT_IN_BEST_CHAIN_CONFIRMATIONS)
+                        }
+                        _ => unreachable!("unmatched response to a depth request"),
+                    }
+                } else {
+                    // TODO: make Depth support heights as well
```

Comment:
Could we add this to a follow-up work section and/or create an issue for it?

Reply:
I'll add it to the follow-up work.

Is there a specific RPC client that needs this? That helps us know if we need a ticket, or if a TODO is enough.

Reply:
> Is there a specific RPC client that needs this?

Not that I know of, a TODO should be enough.